src/compiler/utilities.rs

```diff
@@ -54,6 +60,24 @@ fn get_error_span_for_node<TNode: NodeInterface>(
     create_text_span_from_bounds(pos, error_node.end())
 }
 
+pub fn is_property_name_literal<TNode: NodeInterface>(node: &TNode) -> bool {
+    match node.kind() {
+        SyntaxKind::Identifier
+        | SyntaxKind::StringLiteral
+        | SyntaxKind::NoSubstitutionTemplateLiteral
+        | SyntaxKind::NumericLiteral => true,
+        _ => false,
+    }
+}
+
+pub fn get_escaped_text_of_identifier_or_literal(node: Rc<Node>) -> __String {
+    if is_member_name(&*node) {
+        node.as_member_name().escaped_text()
```

Comment:
I like this `node.as_*()` pattern, eg I exposed a `.as_member_name()` method on `Node` that panics unless it can unwrap the `Node` into one of the AST node sub-variants that implements `MemberNameInterface` (and it returns a `&dyn MemberNameInterface`). It's a little bit like what I was thinking of doing via a macro (eg `unwrap_ast_node!(node, BinaryExpression)`) for unwrapping to concrete AST node types but for unwrapping to a certain AST node interface instead